app/components/email-item-details/component.js

```diff
@@ -1,4 +1,25 @@
+// model: function (params) {
+//   return this.store.findRecord("email", params.email_id);
+// },
+
 import Ember from "ember";
+import { task } from "ember-concurrency";
+const { set, get } = Ember;
 
 export default Ember.Component.extend({
+  init() {
+    this._super(...arguments);
+    if (!this.get("item")) {
```

Comment:
just like real world scenario, lets add few additional attributes in the email get api and in that scenario we may not be able to consume item property as is.. This details component should always hit an api to get complete detials of a given item.